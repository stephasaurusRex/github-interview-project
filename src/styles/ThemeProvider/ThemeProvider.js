import PropTypes from "prop-types";
import React, { Component } from "react";
import { ThemeProvider as Provider } from "styled-components";

import pop from "../themes/pop";
import random from "../themes/random";

/* adds a function to set the theme on top of styled components themeprovider */

class ThemeProvider extends Component {

  state = {
    themes: [pop, random],
    selectedTheme: pop,
  }

  static childContextTypes = {
    themes: PropTypes.array.isRequired,
    selectedTheme: PropTypes.object.isRequired,
    setTheme: PropTypes.func.isRequired,
  }

  getChildContext = () => {
    const { themes, selectedTheme } = this.state;
    return { themes, setTheme: this.setTheme, selectedTheme };
  }

  setTheme = (theme) => {
    this.setState({ selectedTheme: theme });
  }

  render() {
    return <Provider theme={this.state.selectedTheme}>
      {this.props.children}
    </Provider>;
  }
}

/* wrapper for any component that wishes to set the theme */

const themeWrapper = (ComponentToWrap) => class extends Component {

  static contextTypes = {
    themes: PropTypes.array,
    setTheme: PropTypes.func,
    selectedTheme: PropTypes.object,
  }

  render() {
    const { themes, setTheme, selectedTheme } = this.context;
    return (<ComponentToWrap
      {...this.props}
      setTheme={setTheme}
      themes={themes}
      selectedTheme={selectedTheme} />);
  }
};

export default ThemeProvider;
export { themeWrapper };
