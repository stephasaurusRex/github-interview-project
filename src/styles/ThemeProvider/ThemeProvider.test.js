import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import ThemeProvider from "./ThemeProvider";

describe("ThemeProvider", () => {

  it("renders without props", () => {
    const tree = renderer.create(<ThemeProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
