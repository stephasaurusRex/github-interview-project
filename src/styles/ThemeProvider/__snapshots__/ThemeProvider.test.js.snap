// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider renders without props 1`] = `null`;
